src/main/java/org/truffleruby/core/cast/SplatCastNode.java

```diff
@@ -50,6 +52,10 @@ public SplatCastNode(NilBehavior nilBehavior, boolean useToAry) {
         conversionMethod = getContext().getSymbolTable().getSymbol(name);
     }
 
+    public void doNotCopy() {
+        copy = false;
```

Comment:
Should the node be invalidated if the value of copy changes?

Reply:
I see it's only used in the translator so probably not an issue.

Reply:
It's done before it's executed, so no.
It would be ideal to make `copy` a constructor argument but that's complicated as we would need to do some manual translation for arguments to handle this special case.